sensor/common/collector/service_impl.go

```diff
@@ -0,0 +1,133 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// CollectorService is the struct that manages the collector configuration
+type serviceImpl struct {
+	sensor.UnimplementedCollectorServiceServer
+
+	collectorC chan *sensor.MsgToCollector
+
+	connectionManager *connectionManager
+}
+
+func (s *serviceImpl) Notify(e common.SensorComponentEvent) {
+}
+
+func (s *serviceImpl) Start() error {
+	s.collectorC = make(chan *sensor.MsgToCollector)
```

Comment:
I see the `make` here (in the context of my previous comment in `sensor.go`. This channel seems to be used only in this impl, so let's make the channel internal to the `serviceImpl` and remove it from `sensor.go`.

See my suggestion in the `func NewService() Service`

Reply:
Another issue is that we are not closing this channel. This is not easy to do in this case, but the main candidate for that is the `Stop` function (after making sure that the grpc connection is closed and there will be no more messages).

Reply:
I have made the channel internal to `serviceImpl` and close the channel in `Stop`.